### DISCUSSION TITLE
GitHub Actions workflow started failing because of stalled job with no errors

### DISCUSSION THREAD

**Discussion by ymzayek** [community/community#39836] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, I am testing an Actions workflow on my fork to build documentation. I set everything up and merged it on main and it worked just fine. I also had a nightly scheduled build that also has been working. However, now it started to fail because the build is stalling and the max timeout time I set is being reached. I know that this build takes about ~2 hours and I set the timeout to be 6 hours. I cannot see a reason why the build is stalling for so long and there are no error messages in the logs. Any insight to what could cause this? Could it be related to any recent updates on the VM (Ubuntu-latest) that I'm using?

And the last successful build: https://github.com/ymzayek/nilearn/actions/runs/3511946474 (Nov. 21)
And the failed build 1 days later: https://github.com/ymzayek/nilearn/actions/runs/3520878217/jobs/5902188512

Access to fork: https://github.com/ymzayek/nilearn

No merges on main have been done since this was implemented successfully.

**Comment by david-engelmann** [comment]
It seems like the issue occurs in the following step

```
      - name: Build docs
        run: |
          source activate testenv
          echo "Conda active env = $CONDA_DEFAULT_ENV";
          cd doc;
          set -o pipefail;
          PATTERN=$(cat ../pattern.txt) make $(cat ../build.txt) 2>&1 | tee log.txt;
```

I would suggest changing it to `... make -j8 $(cat ...`

**Reply by ymzayek** [reply]
Hi, thanks for looking into it. But I don't think that is the problem, and in fact we cannot parallelize our build because of extensive cross-referencing that we use when building our examples. I know that the build takes ~2 hours (I know this from local builds, CircleCI runs in the original repository: https://github.com/nilearn/nilearn, as well as successful runs in Actions before 2 days ago). The workflow on Actions is just a migration of what is done on CircleCI. The only thing that I just noticed has changed recently is that 'ubuntu-latest' is now using `Ubuntu-22.04` instead of `20.04`. And the example where it stalls opens an html file in the browser so I wonder if it's related to the way I set be the display server. It seems that setting the job to run on `ubuntu-20.04` has resolved the problem for now but I still wonder if my setup is less than optimal.

**Comment by david-engelmann** [comment, marked as answer]
I've seen a few people reporting slower speed after upgrading from 20.04 to 22.04. One suggestion online was the following

```
in /etc/default/grub: Add the flag intel_iommu=off to GRUB_CMDLINE_LINUX_DEFAULT
```

[More info on it can be found here](https://askubuntu.com/questions/1406263/ubuntu-22-04-runs-too-slow)

**Reply by ymzayek** [reply]
Ok I see, that's helpful to know. Thanks!

**Reply by david-engelmann** [reply]
No Problem! Don't forget to close it out!
